### DISCUSSION TITLE
Difference between 'stop()' and 'cancel()' methods

### DISCUSSION THREAD

**Discussion by SebasRebazCoding** [llfbandit/record#363] [Q&A] [answered]
Hey there! I'm pretty new to using this package. I like how simple and straightforward the instructions are, that helps a lot, but I have some doubts. I'm implementing a simple recorder and I'm trying to understand the difference between using stop() and cancel().

I am aware that the documentations detail that 'stop' releases the audio resource after invoking it and 'cancel' does this, while also deleting the recorded file/blob. This brought some questions to my mind and I turn to the community for answers today. Any help is appreciated!

1. If I were to use 'stop' and then start recording again with the same "path" parameter, does it overwrite the already generated file?
2. Is there any benefit to using 'cancel' or 'stop' instead of the other?

**Comment by llfbandit** [comment, marked as answer]
Yes, recording on the same path should overwrite the previous file.
`cancel` method is similar to `stop` plus the deletion of the file/blog recorded.

**Reply by SebasRebazCoding** [reply]
Thank you for the quick answer!
